### PR TITLE
Add server settings

### PR DESCRIPTION
**Pull request by jstzwj**
ServerSettingsGrain, SettingsModule, ServerSettings

### CODE REVIEW

File: src/MineCase.Gateway/Program.cs [hunk 1]
```diff
@@ -56,11 +56,17 @@ private static IConfiguration LoadConfiguration()
...
+            {
+                logger.LogInformation("Connecting to cluster...");
+                await _clusterClient.Connect();
+                logger.LogInformation("Connected to cluster.");
+            }
+            catch (Exception e)
+            {
+                logger.LogError("Cluster connection failed.\n" + "Exception stack trace:" + e.StackTrace);
+                _exitEvent.Set();
+            }
```
Comment (sunnycase) [inline src/MineCase.Gateway/Program.cs]: there should be a `return;`.

File: src/MineCase.Server.Grains/Settings/ServerSettingsGrain.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        public override async Task OnActivateAsync()
+        {
+            string settingsFile = await ReadSettingsAsString("server.json");
+            try
+            {
+                _settings = JsonConvert.DeserializeObject<ServerSettings>(settingsFile);
+            }
+            catch (Exception e)
+            {
+                System.Console.WriteLine("Json deserialization failed\n" + e.StackTrace);
```
Comment (sunnycase) [inline src/MineCase.Server.Grains/Settings/ServerSettingsGrain.cs]: Use `ILoggerFactory `to get a `ILogger`, then use it to log exception.

File: src/MineCase.Server.Grains/Settings/ServerSettingsGrain.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            _settings = settings;
+            return Task.CompletedTask;
+        }
+
+        private Task<string> ReadSettingsAsString(string path)
+        {
+            string result = null;
+            try
+            {
+                result = File.ReadAllText(path);
```
Comment (sunnycase) [inline src/MineCase.Server.Grains/Settings/ServerSettingsGrain.cs]: Use `StreamReader.ReadToEndAsync` instead.

File: src/MineCase.Server.Interfaces/Game/OpPermissionLevel.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace MineCase.Server.Game
+{
+    /// <summary>
+    /// 1 - Ops can bypass spawn protection.
+    /// 2 - Ops can use all singleplayer cheats commands(except /publish, as it is not on servers; along with /debug) and use command blocks.Command blocks and Realms operators/owners have this level.
+    /// 3 - Ops can use most multiplayer-exclusive commands (all except level 4 commands).
+    /// 4 - Ops can use all commands including /stop, /save-all, /save-on, and /save-off.
+    /// </summary>
+    public enum OpPermissionLevel : byte
```
Comment (sunnycase) [inline src/MineCase.Server.Interfaces/Game/OpPermissionLevel.cs]: Does "Op" means "Operation"?

Comment (jstzwj) [inline src/MineCase.Server.Interfaces/Game/OpPermissionLevel.cs]: It means 'operator'.

Comment (sunnycase) [inline src/MineCase.Server.Interfaces/Game/OpPermissionLevel.cs]: So why not use the full name.

Comment (jstzwj) [inline src/MineCase.Server.Interfaces/Game/OpPermissionLevel.cs]: It an acronym in minecraft. The command '\op' is the origin of it. I think most minecraft players know it, who join mc multi player servers.
https://minecraft-zh.gamepedia.com/index.php?search=op&title=Special:%E6%90%9C%E7%B4%A2&fulltext=%E6%90%9C%E7%B4%A2&searchToken=ejcq585adjnbf6a1gms3vzd47

**Pull request merged**